src/nfa/inclusion.cc

```diff
@@ -160,20 +172,27 @@ bool mata::nfa::algorithms::is_included_antichains(
             for (const State& smaller_succ : smaller_move.targets) {
                 const ProdStateType succ = {smaller_succ, bigger_succ, min_dst(bigger_succ)};
 
-                if (lengths_incompatible(succ) || (smaller.final[smaller_succ] &&
-                    !bigger.final.intersects_with(bigger_succ)))
+                if (lengths_incompatible(succ) ||
+                    (smaller.final[smaller_succ] && !bigger.final.intersects_with(bigger_succ)))
                 {
                     if (cex != nullptr) {
-                        cex->word.clear();
                         cex->word.push_back(smaller_symbol);
+                        cex->path.push_back(smaller_state);
                         ProdStateType trav = prod_state;
-                        while (paths[trav].first != trav)
+                        while (paths.at(trav).first != trav)
                         { // go back until initial state
-                            cex->word.push_back(paths[trav].second);
-                            trav = paths[trav].first;
+                            cex->word.push_back(paths.at(trav).second);
+                            cex->path.push_back(std::get<0>(paths.at(trav).first));
+                            trav = paths.at(trav).first;
```

Comment:
Maybe extract the repeated call to `paths.at(trav)` to omit repeated binary searching (`std::map` searches are expensive).

Reply:
Done

Reply:
Looks good.